src/compiler/checker.ts

```diff
@@ -349,6 +349,13 @@ module ts {
                             }
                             result = undefined;
                         }
+                        else if (location.kind === SyntaxKind.SourceFile) {
+                            result = getSymbol(getSymbolOfNode(location).exports, "default", meaning & SymbolFlags.ModuleMember);
+                            if (result && (result.flags & meaning) && result.valueDeclaration && (result.valueDeclaration.flags & NodeFlags.Default) && result.valueDeclaration.localSymbol.name === name) {

```

Comment:
what about adding function to avoid duplicating the same checks in two places?
```ts
function getLocalSymbolForExportDefault(symbol: Symbol) {
 return symbol.valueDeclaration && (symbol.valueDeclaration.flags & NodeFlags.Default) && symbol.valueDeclaration.localSymbol ? symbol.valueDeclaration.localSymbol : undefined;
}
```